lms/djangoapps/courseware/module_render.py

```diff
@@ -1068,31 +1068,32 @@ def handle_xblock_callback(request, course_id, usage_id, handler, suffix=None):
         return _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course=course)
 
 
-def get_module_by_usage_id(request, course_id, usage_id, disable_staff_debug_info=False, course=None,
-                           will_recheck_access=False):
+def _get_usage_key_for_course(course_key, usage_id) -> UsageKey:
     """
-    Gets a module instance based on its `usage_id` in a course, for a given request/user
-
-    Returns (instance, tracking_context)
+    Returns UsageKey mapped into the course for a given usage_id string
     """
-    user = request.user
-
     try:
-        course_id = CourseKey.from_string(course_id)
-        usage_key = UsageKey.from_string(unquote_slashes(usage_id)).map_into_course(course_id)
-    except InvalidKeyError:
-        raise Http404("Invalid location")  # lint-amnesty, pylint: disable=raise-missing-from
+        return UsageKey.from_string(unquote_slashes(usage_id)).map_into_course(course_key)
+    except InvalidKeyError as exc:
+        raise Http404("Invalid location") from exc
 
+
+def _get_descriptor_by_usage_key(usage_key):
```

Comment:
The surrounding little refactor was all to just expose `_get_usage_key_for_course` and `_get_descriptor_by_usage_key` so that we could call them outside the context of `get_module_by_usage_id`.